src/borg/archiver.py

```diff
@@ -3025,7 +3025,10 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
             group.add_argument('-a', '--glob-archives', metavar='GLOB', dest='glob_archives',
                                type=GlobSpec, action=Highlander,
                                help='only consider archive names matching the glob. '
-                                    'sh: rules apply, see "borg help patterns".')
+                               'sh: rules apply (without actually using the sh: prefix), see "borg help patterns".\n'
+                               'Warning: In borg v2 the option -a will accept general patterns like sh:, re:, etc.'
+                               'This has the sideffect that patterns with a wildcard (*) will not be expanded but matched exactly.'
+                               'See borg v2 documentation for borg prune.')
```

Comment:
I don't think we should talk about borg2 here, such stuff is part of the changelog / upgrade notes section there.

Also maybe don't change the indentation, not sure if the style checker is happy with that.

Reply:
Sorry, I didn't let this run through CI. Will do it as soon as I have time.

Reply:
btw, this is from the upgrade notes section in changes.rst:

```
the --glob-archives option was renamed to --match-archives (the short option name -a is unchanged) and extended to support different pattern styles:

id: for identical string match (this is the new default!)
sh: for shell pattern / globbing match (this was used by --glob-archives)
re: for regular expression match
```

Reply:
> btw, this is from the upgrade notes section in changes.rst:
Thanks for pointing this out.

> Also maybe don't change the indentation, not sure if the style checker is happy with that.
I will change it, although the linter seems to be happy.

Reply:
Ok I deleted the future warning and made the indentation to be identical to before.